### PR TITLE
Feature/java integration tests

### PR DESCRIPTION
**Pull request by vanajamukkara**
Added readme.md

Added a script which gets executed at post-integration-test phase to take the dump of pods, domain, etc logs after the test completes and creates a test out file under RESULT_ROOT dir. Tests need to be called with "mvn clean verify -P java-integration-tests" to get the post-integration-test phase executed.

### CODE REVIEW

File: integration-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
+# Integration Tests for Operator
+
+This documentation describes the functional use cases that are covered in integration testing for Weblogic Operator. The tests are written in java(Junit tests) and driven by maven profile. All the use cases covered in run.sh will be covered in java integration tests. Currently, QUICKTEST use cases and some of non-quick test(till domain4) use cases are covered in java tests.
```
Comment (rjeberhard) [inline integration-tests/README.md]: Full name is "the Oracle WebLogic Server Kubernetes Operator".   

"java" -> "Java"
"Junit" -> "JUnit"
"maven" -> "Maven"

Need a space after java in second line.

File: integration-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+# Environments
+
+The tests currently runs in three modes, "Wercker", "Jenkins", and "standalone" Oracle Linux, where the mode is controlled by the WERCKER and JENKINS environment variables described below. The default is "standalone".
```
Comment (rjeberhard) [inline integration-tests/README.md]: currently run in three modes:

File: integration-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+* create operator operator1 which manages default and test1 namespaces, verify its deployed successfully, pod created, operator Ready and verify external REST service if configured
+* create domain domain1 in default namespace and verify the pods, services are created and servers are in Ready
+* verify admin external service by accessing admin REST endpoint with nodeport in URL
+* verify admin t3 channel port by exec into the admin pod and deploying webapp using the channel port for WLST
+* verify web app load balancing  
+* verify domain life cycle(destroy and create) should not any impact on Operator managing the domain and web app load balancing and admin external service
+* cluster scale up/down using Operator REST endpoint, webapp load balancing should adjust accordingly. (run.sh does scaling by editing the replicas in domain-custom-resource.yaml.)
+* Operator life cycle(destroy and create) should not impact the running domain
+
+Non-quick test use cases
```
Comment (rjeberhard) [inline integration-tests/README.md]: Non-quick?  :). Can we call if "Full test"?

File: integration-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+namespace= default
+javaOptions= -Dweblogic.StdoutDebugEnabled=false
+
+```
+Certain properties like weblogicDomainStoragePath, weblogicOperatorImage, externalSans are populated at run time.
+
+
+# How does it work
+
+When the tests are run manually with mvn command on hosted linux, weblogic image and server jre images are pulled from a local repository. Operator image is built with the git branch from where the mvn command is executed.
```
Comment (rjeberhard) [inline integration-tests/README.md]: WebLogic, Linux, Java

File: integration-tests/README.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+All the tests that start with IT*.java are ran. The test builds the operator, runs a series of tests and archives the results into tar.gz files upon completion.
```
Comment (rjeberhard) [inline integration-tests/README.md]: are run.

File: integration-tests/README.md (hunk 1 above)
Comment (vanajamukkara) [reply integration-tests/README.md]: Done

File: integration-tests/README.md (hunk 2 above)
Comment (vanajamukkara) [reply integration-tests/README.md]: Done

File: integration-tests/README.md (hunk 3 above)
Comment (vanajamukkara) [reply integration-tests/README.md]: Changed to Full test

File: integration-tests/README.md (hunk 4 above)
Comment (vanajamukkara) [reply integration-tests/README.md]: Done

File: integration-tests/README.md (hunk 5 above)
Comment (vanajamukkara) [reply integration-tests/README.md]: Done

**Pull request merged**